flood-fill.py

```diff
@@ -1,14 +1,89 @@
+from pprint import pprint
 grid = [
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
 ]
 
+
 def fill(row, col, new_color):
-    return # TODO practice for geoff
\ No newline at end of file
+        global grid
+        old_color = grid[row][col]
+        navigate(row, col, new_color, old_color)
+        return grid
+
+
+def navigate(row, col, new_color, old_color):
+        global grid
+        print("Checking grid[{}][{}]".format(row, col))
+        directions = {"north": ["row", -1], "south": ["row", 1], "east": ["col", 1], "west": ["col", -1]}
```

Comment:
there's much shorter ways to accomplish this:
directions = [(0,1), (0,-1), (1, 0), (-1, 0)] - then you can treat each case as the same and not spend so much code on if/else - also if you add more directions (like northwest) you won't have to write any more code

Reply:
I set it up as a dicitonary mostly to help myself visualize it. (I guess that's what comments are for though, lol). Going to try to figure out how to make your suggestion work.